libyara/ahocorasick.c

```diff
@@ -811,6 +811,7 @@ int yr_ac_automaton_create(
 int yr_ac_automaton_destroy(
     YR_AC_AUTOMATON* automaton)
 {
+  if (automaton == NULL) return ERROR_SUCCESS;
```

Comment:
Return `ERROR_INVALID_ARGUMENT` instead.

Reply:
Done, I was unsure this would have side effect since it's a failure code.